src/coreclr/src/vm/processdiagnosticsprotocolhelper.cpp

```diff
@@ -0,0 +1,210 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "common.h"
+#include "fastserializer.h"
+#include "processdiagnosticsprotocolhelper.h"
+#include "eventpipeeventsource.h"
+#include "diagnosticsprotocol.h"
+
+#ifdef FEATURE_PERFTRACING
+
+static bool IsNullOrWhiteSpace(LPCWSTR value)
+{
+    if (value == nullptr)
+        return true;
+
+    while (*value)
+    {
+        if (!iswspace(*value))
+            return false;
+        ++value;
+    }
+    return true;
+}
+
+static inline uint32_t GetStringLength(const char *&value)
+{
+    return static_cast<uint32_t>(strlen(value) + 1);
+}
+
+static inline uint32_t GetStringLength(const WCHAR *&value)
+{
+    return static_cast<uint32_t>(wcslen(value) + 1);
+}
+
+template <typename T>
+static bool TryWriteString(uint8_t * &bufferCursor, uint16_t &bufferLen, const T *&value)
```

Comment:
Similar to comment above, but do we need this to be a template if all the calls to `TryWriteString` are happening with `LPCWSTR` types?

Reply:
Same as my previous comment.  I'll remove the templating for these.